crates/anemo/src/crypto.rs

```diff
@@ -129,10 +131,9 @@ impl rustls::client::ServerCertVerifier for ExpectedCertVerifier {
         let peer_id = peer_id_from_certificate(end_entity)?;
 
         if peer_id != self.1 {
-            return Err(rustls::Error::InvalidCertificateData(format!(
-                "invalid peer certificate: received {:?} instead of expected {:?}",
-                peer_id, self.1,
-            )));
+            return Err(rustls::Error::InvalidCertificate(
+                rustls::CertificateError::NotValidForName,
```

Comment:
Is there a way to keep the formatted error? The text has been very useful in debugging things, event today

Reply:
PTAL, it's kinda ugly but I made it work